nix/keys.nix

```diff
@@ -15,11 +15,33 @@ let
         <filename><replaceable>destDir</replaceable>/<replaceable>password</replaceable></filename>
         will be <literal>foobar</literal>.
 
-        NOTE: Either <literal>text</literal> or <literal>keyFile</literal> have
-        to be set.
+        NOTE: Either <literal>text</literal>, <literal>keyCmd</literal> or
+        <literal>keyFile</literal> have to be set.
       '';
     };
 
+    options.keyCmd = mkOption {
+      default = null;
+      example = "pass show secrettoken";
+      type = types.nullOr types.str;
```

Comment:
What if this was a listof str's, which are executed without a shell?

Reply:
@grahamc well I could do that thing but I'm not sure how to process it on Python side. I mostly copy-pasted python code for this from other options :3 need to study the internal APIs more I guess?